### DISCUSSION TITLE
hierarchical noun_chunks

### DISCUSSION THREAD

**Discussion by Fan-Luo** [explosion/spaCy#8043] [New Features & Project Ideas]
Hi Team,

I wonder would you consider add hierarchical noun_chunks extraction feature.

For example:
```
text ='"Tomb Raider III" is the third title in the "Tomb Raider" video game series and a sequel to "Tomb Raider II".'
doc = nlp(text)
for chunk in doc.noun_chunks:
    print('chunk: ', chunk.text) 
```
print out:
> chunk:  Tomb Raider III
> chunk:  the third title
> chunk:  the "Tomb Raider" video game series
> chunk:  a sequel
> chunk:  "Tomb Raider II

I was expecting `Tomb Raider` itself would also be a noun_chunk.

Best

**Comment by polm** [comment]
The noun chunks in spaCy are defined in a simple way, as a series of adjectives and nouns (with incidental punctuation). This definition is unambiguous and makes it easy to guess what the output should be a from a given sentence.

How would you specify your hierarchical noun chunks? What would the output be for "scary wolf man" or "fastest hot dog eater", and how would you define the nested chunks in code? It doesn't seem straightforward to me. 

If you do have a way to describe the chunks you want you should be able to do it with the Matcher though.

**Reply by Fan-Luo** [reply]
Thank you for your reply!

I can understand the intention, similar as fuzzywuzzy's [dedupe](https://github.com/seatgeek/fuzzywuzzy/blob/9e3d2fe0d8c1b195696d5fbcda78c371dd4a6b8f/fuzzywuzzy/process.py#L225), when deduplicate a list of similar strings, looks for the longest item in the duplicate list, because it contains the most information.

I am not sure did you break the word `wolfman` and `hotdog` on purpose. 
In these two simple cases, I would like `wolfman` and `hotdog` to also be extracted as a 'lower-level' noun_chunk.  This can be simply obtained with `chunk.root`. 

But in a more complex phrase, say `the third title in the "Tomb Raider" video game series`, the 'high-level' noun_chunk contains several lower-level noun_chunks `the third title`, `Tomb Raider`, and `video game series`, which can not be easily  accessed.

Best Regards
